apps/gallery/js/ImageEditor.js

```diff
@@ -371,10 +400,14 @@ function setEditTool(editTool) {
       showEditToolView();
       $('edit-crop-options').classList.remove('hidden');
       $('edit-title').setAttribute('data-l10n-id', 'crop');
-      // Show crop overlay on entering crop edit tool
-      imageEditor.edit(function() {
-        imageEditor.showCropOverlay();
-      });
+        // Initialize the source rectangle to full size of the original image
+        imageEditor.resetCropRegion();
+        // Show screen with crop overlay on entering crop edit tool
+        imageEditor.requestNewPreview(function() {
+          // Use saved crop region in editSettings to show crop overlay
+          imageEditor.showCropOverlay(
+            imageEditor.getCropOverlayRegion(savedEditSettings.crop));
```

Comment:
requestNewPreview generates and displays new preview, and updates editSettings.crop in generateNewPreview https://github.com/punamdahiya/gaia/blob/Bug1087494/apps/gallery/js/ImageEditor.js#L829
So at this point values in editSettings and savedEditSettings are different and we need to use  savedEditSettings.crop to show crop overlay with cropped size from last edit.